RTSP-Stream-Video-Capturer/main.cpp

```diff
@@ -45,18 +45,17 @@ static otk_thread_func_return_type capturer_thread_start_function(void *arg) {
   cv::Mat opencv_frame;
   while((video_capturer->capturer_thread_exit.load() == false) &&
         opencv_video_capture.read(opencv_frame)) {
-    cv::cvtColor(opencv_frame, opencv_frame, cv::COLOR_BGR2YUV);
-    std::vector<cv::Mat> yuv;
-    cv::split(opencv_frame, yuv);
-    otc_video_frame *otc_frame = otc_video_frame_new_I420(
-        video_capturer->width,
-        video_capturer->height,
-        yuv[0].ptr<uchar>(),
-        yuv[0].step[0],
-        yuv[1].ptr<uchar>(),
-        yuv[1].step[0],
-        yuv[2].ptr<uchar>(),
-        yuv[2].step[0]);
+    cv::cvtColor(opencv_frame, opencv_frame, cv::COLOR_BGR2RGB);
+    const uchar* planes[1] {opencv_frame.ptr<uchar>()};
+    int strides[1] {static_cast<int>(opencv_frame.step[0])};
+
+    otc_video_frame *otc_frame = otc_video_frame_new_from_planes(
```

Comment:
This was a bit tricky. This is the best way I could find to convert opencv frames to opentok ones. The previous code was leading to a video with somehow corrupted colours (maybe an issue with the I420 chroma downsampling?).

Reply:
I think it is just 4:2:0 is the most common one. If I am not wrong that's the format we use to inject frames into webrtc. Of course when we use a different one, another (transparent) conversion is needed. 

It seems that the video from opencv capturer is coming in BGR format. So a conversion is needed anyway, we do not have any function in OTKit to take a BGR frame (it could be added of course, but that's not the purpose of this ticket).

I tried using OTC_VIDEO_FRAME_FORMAT_YUV420P (actually that was the one used before). It is providing a corrupted video in terms of colours. My guess is that the previous conversion `COLOR_BGR2YUV` was not providing downsampled UV planes as our `otc_video_frame_new_*` functions expect for 420P. Of course we could have fixed it with some additional operation after conversion, but I think it is not worth for a sample app that has a completely different purpose (RTSP). And  this approach works without further processing. 

OpenCV also has a `COLOR_BGR2YUV_I420` that may work. But it seems you need to manually separate the planes. Again, for a sample app I prefer not to introduce such code.

Reply:
Completely agree. Thank you for the clarification! Once the PR is building is an ok from me